awx/main/consumers.py

```diff
@@ -176,9 +191,20 @@ async def receive_json(self, data):
                     self.channel_name,
                 )
 
+            if len(old_groups):
+                await self.channel_layer.group_send(
+                    settings.BROADCAST_WEBSOCKET_GROUP_NAME,
+                    {"type": "consumer.unsubscribe", "groups": list(old_groups), "origin_channel": self.channel_name},
+                )
+
             new_groups_exclusive = new_groups - current_groups
             for group_name in new_groups_exclusive:
                 await self.channel_layer.group_add(group_name, self.channel_name)
+
+                await self.channel_layer.group_send(
```

Comment:
Just making a note for my future self: I think this line is meant to be de-dented one level.